File: compiler-solo-ast/src/test/resources/IntUnsignedShiftRight.java
public class IntUnsignedShiftRight {
    public static void main(String[] args) {
        StaticJavaLib.assertTrue( 2 >>> 1 == 1);
        StaticJavaLib.assertTrue(-2 >>> 1 == 2147483647);

        StaticJavaLib.assertTrue(unsignedShiftRight( 2, 1) == 1);
        StaticJavaLib.assertTrue(unsignedShiftRight(-2, 1) == 2147483647);
    }
    
    static int unsignedShiftRight(int i, int j){
        return i >>> j;
    }
}

File: compiler-group-traslator/src/test/resources/c/IntUnsignedShiftRight.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include "GC.c"
#include "ExceptionHandlingFunctions.c"
VM* vm;
int unsignedShiftRight(int i, int j) {
	return ((unsigned int)i) >> j;

	//set all vars to zero and have the TEMP variable for the return expression
}
int main(char* args) {
	vm = newVM();
	assertTrue(((unsigned int)2) >> 1 == 1);
	assertTrue(((unsigned int)-2) >> 1 == 2147483647);
	assertTrue(unsignedShiftRight(2, 1) == 1);
	assertTrue(unsignedShiftRight(-2, 1) == 2147483647);
	return 0;
}